src/main/kotlin/com/exactpro/th2/codec/AbstractCodecProcessor.kt

```diff
@@ -35,13 +36,16 @@ abstract class AbstractCodecProcessor(
     protected val codec: IPipelineCodec,
     private val codecEventID: EventID,
     private val onEvent: (event: ProtoEvent) -> Unit,
-) : MessageProcessor<GroupBatch, GroupBatch> {
+) {
     private val logger = KotlinLogging.logger {}
 
     protected fun onEvent(message: String, messagesIds: List<MessageID> = emptyList()) = codecEventID.onEvent(message, messagesIds)
 
     protected fun onErrorEvent(message: String, messagesIds: List<MessageID> = emptyList(), cause: Throwable? = null) = codecEventID.onErrorEvent(message, messagesIds, cause)
 
+    abstract fun process(source: GroupBatch): GroupBatch
+    abstract fun process(source: MessageGroupBatch): MessageGroupBatch
```

Comment:
Please split logic to two separate classes to simplify deprecation process in future